ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -64,17 +70,29 @@ public DataStreamClientImpl(RaftPeer raftServer,
     this.orderedStreamAsync = new OrderedStreamAsync(dataStreamClientRpc, properties);
   }
 
-  class DataStreamOutputImpl implements DataStreamOutput {
+  public class DataStreamOutputImpl implements DataStreamOutput {
     private long streamId = 0;
     private long messageId = 0;
+    private RaftClientRequest request;
 
     public DataStreamOutputImpl(long id){
       this.streamId = id;
     }
 
     // send to the attached dataStreamClientRpc
     @Override
-    public CompletableFuture<DataStreamReply> streamAsync(ByteBuffer buf) {
+    public CompletableFuture<DataStreamReply> streamRequestAsync() {
+      messageId++;
+      RaftClientRequest.Type type = RaftClientRequest.dataStreamRequestType(streamId, messageId, false);
```

Comment:
Let's use RaftClientRequest.writeRequestType() for the moment since the effect of dataStream should be the same as AsyncApi.send.  If there is a need, we may add dataStreamRequestType later.